app/src/main/java/cm/aptoide/pt/account/view/LoginSignUpCredentialsFragment.java

```diff
@@ -320,9 +334,38 @@ private void setAptoideSignUpLoginAreaVisible() {
     }
     loginArea = view.findViewById(R.id.login_button_area);
     signUpArea = view.findViewById(R.id.sign_up_button_area);
-    termsAndConditions = (TextView) view.findViewById(R.id.terms_and_conditions);
     separator = view.findViewById(R.id.separator);
 
+    ClickableSpan clickableSpan = new ClickableSpan() {
+      @Override public void onClick(View view) {
+        termsAndConditionsClicked.onNext(null);
+      }
+    };
+
+    ClickableSpan clickableSpan2 = new ClickableSpan() {
+      @Override public void onClick(View view) {
+        privacyPolicyClicked.onNext(null);
+      }
+    };
+
+    String baseString = getString(R.string.terms_and_conditions_privacy_sign_up_message);
+    String termsAndConditionsPlaceHolder = getString(R.string.settings_terms_conditions);
+    String privacyPolicyPlaceHolder = getString(R.string.settings_privacy_policy);
+    String compositeString =
+        String.format(baseString, termsAndConditionsPlaceHolder, privacyPolicyPlaceHolder);
+
+    SpannableString compositeSpan = new SpannableString(compositeString);
+    compositeSpan.setSpan(clickableSpan, compositeString.indexOf(termsAndConditionsPlaceHolder),
+        compositeString.indexOf(termsAndConditionsPlaceHolder)
+            + termsAndConditionsPlaceHolder.length(), Spanned.SPAN_EXCLUSIVE_EXCLUSIVE);
+    compositeSpan.setSpan(clickableSpan2, compositeString.indexOf(privacyPolicyPlaceHolder),
+        compositeString.indexOf(privacyPolicyPlaceHolder) + privacyPolicyPlaceHolder.length(),
+        Spanned.SPAN_EXCLUSIVE_EXCLUSIVE);
```

Comment:
Maybe put this in a method that receives the String as an argument
Naming compositeString, maybe privacyAndTermsPlaceholderSpan